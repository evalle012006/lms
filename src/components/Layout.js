import { useEffect } from "react";
import { useSelector, useDispatch } from 'react-redux';
import { fetchWrapper } from '@/lib/fetch-wrapper';
import { setCurrentDate } from '@/redux/actions/systemActions';
import NavComponent from "./Nav";

const Layout = ({ children, bgwhite = false, header = true, noPad = false, actionButtons = [], hScroll = true }) => {
    const state = useSelector(state => state.global);
    const pageTitle = state.title;
    const dispatch = useDispatch();

    const getCurrentDate = async () => {
        const apiURL = `${process.env.NEXT_PUBLIC_API_URL}settings/current-date`;
        const response = await fetchWrapper.get(apiURL);
        if (response.success) {
            dispatch(setCurrentDate(response.currentDate));
        }
    }

    useEffect(() => {
        getCurrentDate();
    }, []);

    return (
        <div className="flex bg-white">
            <NavComponent />
            <div className={`ml-[16rem] flex flex-col bg-neutral-200 duration-300 w-screen h-screen ${hScroll ? 'overflow-x-auto' : 'overflow-hidden'}`}>
                {header && (
                    <div className="bg-white p-6 gap-6 h-20">
                        <div className="flex flex-row justify-between">
                            <div className="page-title">
                                {pageTitle}
                            </div>
                            <div className="flex flex-row">
                                {actionButtons && actionButtons.map((btn, idx) => {
                                    return (
                                        <div key={idx} className="mr-2">
                                            {btn}
                                        </div>
                                    );
                                })}
                            </div>
                        </div>
                    </div>
                )}
                <div className={`${bgwhite && 'bg-white'}`}>
                    {children}
                </div>
            </div>
        </div>
    );
}

export default Layout;