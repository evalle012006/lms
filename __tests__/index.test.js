import renderer from 'react-test-renderer';
import Layout from '@/components/Layout';

describe('Hybrid AG Test Suite', () => {
    it('runs just testing correctly', () => {
        expect(true).toEqual(true);
    });

    it('layout renders correctly', () => {
        const tree = renderer
            .create(<Layout />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});