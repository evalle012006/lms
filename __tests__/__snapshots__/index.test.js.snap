// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hybrid AG Test Suite layout renders correctly 1`] = `
<div
  className="container mx-auto pt-20"
>
  <main />
</div>
`;
